### PR TITLE
Blocks: use Jetpack green on all block icons

### PR DESCRIPTION
**Pull request by keoshi**
#### Changes proposed in this Pull Request:

* Color all icons on Jetpack blocks with the green color.

#### Before

image

#### After

image

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* No.

#### Testing instructions:

* Fire up this PR.
* Create a new post/page.
* Add a new Jetpack block either by going to the JP section or searching for one.
* Ensure all the blocks have the green colored icon.

#### Proposed changelog entry for your changes:
* None.

### CODE REVIEW

**Review by jeherve — COMMENTED** [review]
Do you think it would be possible to create a shared function that would turn those icons green from one central place? Doing it that way would allow us to filter and avoid doing it on WordPress.com where that added color may not make as much sense?

**Comment by keoshi** [comment]
@jeherve Good point, I didn't think about dotcom while doing these changes. I know very little about Jetpack blocks in the context of dotcom, but your suggestion sounds great. Is there prior art in using something like `IS_WPCOM` while registering/configuring JP blocks?

**Comment by jeherve** [comment]
>  Is there prior art in using something like IS_WPCOM while registering/configuring JP blocks?

Yes, we have utility functions we could use in a shared function, to limit the change to Jetpack users only:
https://github.com/Automattic/jetpack/blob/8626cc547cf3f8dd91c730d50b7ba5a2e6e11a27/extensions/shared/site-type-utils.js#L12-L28

**Comment by keoshi** [comment]
This should be ready for another review.

File: extensions/blocks/amazon/index.js
Comment (jeherve) [inline extensions/blocks/amazon/index.js:22]: Since we're not using that constant anywhere else in the block files, we probably don't need to export it; you could use `getIconColor()` right away when defining the icon.

File: extensions/blocks/calendly/icon.js
Comment (jeherve) [inline extensions/blocks/calendly/icon.js:12]: I am an SVG amateur. Could you explain what's changed here?

File: extensions/blocks/wordads/index.js
Comment (jeherve) [inline extensions/blocks/wordads/index.js:42]: I think you can update that one too.

File: extensions/shared/block-icons.js
Comment (jeherve) [inline extensions/shared/block-icons.js:15]: Instead of hardcoding the color value, it would be nice if we could pull it from `@wordpress/base-styles` somehow, but I am not sure we can access those scss variables in a js file to begin with. We'll need to remember to keep that value updated. Maybe we could add a comment pointing to where the default Gutenberg black is defined, so we can refer to it and update ours when it changes?

File: extensions/shared/block-icons.js
Comment (jeherve) [inline extensions/shared/block-icons.js:18]: Same as above maybe?
https://github.com/Automattic/jetpack/blob/ed275e2d8167f79899c4965719f76410e244be87/_inc/client/scss/variables/_colors.scss#L95

**Review by jeherve — CHANGES_REQUESTED** [review]

File: extensions/blocks/calendly/icon.js
Comment (keoshi) [reply extensions/blocks/calendly/icon.js:12]: I imported the SVG code into Illustrator and cleaned it up a bunch. Essentially where we were using shapes (rectangles, paths) I combined everything into a single path that can be more easily addressable. In the process, I also removed the inline color definitions.

File: extensions/blocks/amazon/index.js
Comment (keoshi) [reply extensions/blocks/amazon/index.js:22]: Good point — will update those.

File: extensions/shared/block-icons.js
Comment (keoshi) [reply extensions/shared/block-icons.js:15]: I looked this up and it seems like we'd need some sort of [custom webpack configuration](https://til.hashrocket.com/posts/sxbrscjuqu-share-scss-variables-with-javascript) along with an `:export` command on the scss file, both of which I'm unsure how to make or if we can even make them.

Comment (keoshi) [reply extensions/shared/block-icons.js]: Added comments on https://github.com/Automattic/jetpack/pull/15311/commits/7633b59aab9c2d5f44d22679b0b79996f2ad0d1d

File: extensions/shared/block-icons.js
Comment (keoshi) [reply extensions/shared/block-icons.js]: Added comments on https://github.com/Automattic/jetpack/pull/15311/commits/7633b59aab9c2d5f44d22679b0b79996f2ad0d1d

File: extensions/blocks/amazon/index.js
Comment (keoshi) [reply extensions/blocks/amazon/index.js]: Done in https://github.com/Automattic/jetpack/pull/15311/commits/7779dded8f0131ba5101d0f6c04a1d787712b98e

File: extensions/blocks/wordads/index.js
Comment (keoshi) [reply extensions/blocks/wordads/index.js]: Great catch — updated in https://github.com/Automattic/jetpack/pull/15311/commits/3c4c406d2aebc152cb43348ca07a3d1bae96a7e3

**Comment by folletto** [comment]
Design wise: 👍 
Let's see how it feels and review if we get feedback before the next release.

File: extensions/blocks/google-calendar/icon.js
Comment (jeherve) [inline extensions/blocks/google-calendar/icon.js:14]: Tests are failing here:

```
14:82  error  A space is required before closing bracket  react/jsx-tag-spacing
```

File: extensions/blocks/calendly/icon.js
Comment (jeherve) [inline extensions/blocks/calendly/icon.js:8]: Tests are failing here:
```
12:79  error  A space is required before closing bracket  react/jsx-tag-spacing
```

**Review by jeherve — CHANGES_REQUESTED** [review]

**Comment by keoshi** [comment]
Sorry for the extra back and forth, @jeherve — should be resolved per https://github.com/Automattic/jetpack/pull/15311/commits/2d86db504798e99d0838a2e60685b5b320b708d1

File: extensions/blocks/contact-form/index.js
Comment (jeherve) [inline extensions/blocks/contact-form/index.js:26]: Since we don't use `icon` elsewhere, I don't think we need to export it here.

File: extensions/blocks/contact-info/index.js
Comment (jeherve) [inline extensions/blocks/contact-info/index.js:31]: Same here.

File: extensions/blocks/related-posts/index.js
Comment (jeherve) [inline extensions/blocks/related-posts/index.js:17]: Same here.

File: extensions/blocks/simple-payments/index.js
Comment (jeherve) [inline extensions/blocks/simple-payments/index.js:30]: Same here.

File: extensions/blocks/subscriptions/index.js
Comment (jeherve) [inline extensions/blocks/subscriptions/index.js:19]: Same here.

**Review by jeherve — CHANGES_REQUESTED** [review]
This looks good to me. I only have minor nitpicks!

I was also wondering if such a vivid green color on a white background was good accessibility-wise? 

Tools like [this one](https://webaim.org/resources/contrastchecker/) seem to indicate that it wouldn't be okay, but I am no expert:

image

File: extensions/shared/block-icons.js
Comment (scottsweb) [inline extensions/shared/block-icons.js:22]: Does it matter this is not from the color studio?

Comment (keoshi) [reply extensions/shared/block-icons.js:22]: Good point, we should probably fix that. This happened a while ago, but I believe the colors were hardcoded for a specific reason — hence the comments.

Maybe we can do something like this, and call each value as strings:

https://github.com/Automattic/jetpack/blob/ea5269852da76e64683327b7e9fb976188cf4af1/modules/search/instant-search/components/jetpack-colophon.jsx#L8

**Comment by jasmussen** [comment]
The color change is fine to do! 

Note that the green color only achieves 2.5:1 contrast ratio. Of course up to the team to decide whether the label is sufficient, but if you can manage to get it up to 3:1 it reaches AA, which would be a nice quick win. #00AB27 does that.

File: extensions/shared/block-icons.js
Comment (scottsweb) [reply extensions/shared/block-icons.js:22]: if it is done for a reason, that is OK... but the value itself might need to be updated? I think this is the old brighter green perhaps. I couldn't see it anyway.

Comment (folletto) [reply extensions/shared/block-icons.js:22]: Well spotted. We need to use our Jetpack green :)

**Comment by keoshi** [comment]
Thanks for the review, @scottsweb and @joendotcom !

As mentioned by Scott, let's try to use Color Studio here and while we're at it, bump the contrast ratio as suggested by Joen. Using `Jetpack Green 40` will mean having a 3.55:1 ratio.

This change was done in https://github.com/Automattic/jetpack/pull/15311/commits/3cfb5eb05f4e2220edd5bc1707947bf1277598a5

**Comment by scottsweb** [comment]
Screenshot 2020-08-31 at 14 55 39

Getting closer, but I think things have moved on since the original PR - I will see if I can catch them all. Also has me wondering if we want to update the Jetpack SVG.

**Comment by scottsweb** [comment]
Screenshot 2020-08-31 at 15 19 49

I think we have them all. Logos updated too and now using color studio.

**Review by scottsweb — DISMISSED** [review]
Approving 👍

File: extensions/blocks/send-a-message/whatsapp-button/index.js
Comment (jeherve) [inline extensions/blocks/send-a-message/whatsapp-button/index.js:9]: Since we're one level downer, we're going to have to update that path:

```suggestion
import { getIconColor } from '../../../shared/block-icons';
```

File: extensions/blocks/podcast-player/index.js
Comment (jeherve) [inline extensions/blocks/podcast-player/index.js:34]: Let's take that opportunity to update the color here as well?
https://github.com/Automattic/jetpack/blob/update/editor-blocks-icon-color/extensions/blocks/podcast-player/index.js#L86-L87

**Review by jeherve — CHANGES_REQUESTED** [review]

**Comment by keoshi** [comment]
Thanks for updating, Scott! Updating labels again so we can get a final review before merging!

File: extensions/blocks/podcast-player/index.js
Comment (scottsweb) [reply extensions/blocks/podcast-player/index.js:34]: All set

**Review by jeherve — CHANGES_REQUESTED** [review]
The Subscriptions and Whatsapp block icons are still black on my end for some reason, maybe because of hardcoded colors within the icon SVGs?

image

We would also need to add the color to the new Donations block.

I would also recommend adding the pattern to `wp-cli-templates/block-index-js.mustache` for future blocks.

**Comment by scottsweb** [comment]
Good for another check @jeherve

**Review by jeherve — APPROVED** [review]
This looks good. I just committed 136447e94696bee75f94e5c2e88951213f9f5b19 for the Donations block. It should be good to merge now!

**Comment by scottsweb** [comment]
Thanks @jeherve @keoshi  - good teamwork 🤗

**Comment by keoshi** [comment]
Thank **_YOU_**, Scott!

**Comment by jeherve** [comment]
Note that this created an issue with the Form block:
https://github.com/Automattic/jetpack/issues/17089

**Comment by jeherve** [comment]
r213265-wpcom (including changes from #17092)

**Pull request merged**
